### DISCUSSION TITLE
Store NBTCompound in file

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#991] [Help] [answered]
I want to store items to a config. I have tried to get the toItemNBT, store that, and do: ItemStack.fromItemNBT to get the item. The problem is, that I will get errors when I am trying to get the compound. error: https://hastebin.nl/tVb2y9i code: 
![image](https://user-images.githubusercontent.com/71212639/165373339-fb9acb8e-eb84-40e5-baf1-e1cdb5ebefb1.png)
edit: I can store the data in the file.

**Comment by jglrxavpok** [comment, marked as answer]
Hephaistos NBT objects are not easily serialisable to JSON.
Use NBTGsonWriter & NBTGsonReader to read/write NBT from/to JSON.

Also, note that JSON and NBT are **not** interchangeable. You should probably store the SNBT inside the JSON. Or store a binary version next to your inventory.json if you can.

**Reply by DevBlokJochem** [reply]
Uhm, I think that this is too hard for me. can you help me a little bit by giving me an example how to do this?

**Comment by kiip1** [comment]
https://github.com/Minestom/Minestom/discussions/955 Could help you

**Reply by DevBlokJochem** [reply]
tnx
